src/javascript/app_2/App/Components/Layout/Footer/__tests__/toggle_settings.spec.js

```diff
@@ -0,0 +1,61 @@
+import React                  from 'react';
+import { expect }             from 'chai';
+import { configure, shallow } from 'enzyme';
+import Adapter                from 'enzyme-adapter-react-16';
+import { ToggleSettings }     from '../toggle_settings.jsx';
+import { IconSettings }       from 'Assets/Footer';
+import SettingsDialog         from '../../../Elements/SettingsDialog/settings_dialog.jsx';
+import { CSSTransition }      from 'react-transition-group';
+
+configure({ adapter: new Adapter() });
+
+describe('ToggleSettings', () => {
+    it('should render one <ToggleSettings /> component', () => {
+        const wrapper = shallow(<ToggleSettings  />);
+        expect(wrapper).to.have.length(1);
+    });
+    it('should have active class if is_settings_visible is true', () => {
+        const wrapper = shallow(<ToggleSettings is_settings_visible={true} />);
+        expect(wrapper.find('.active').exists()).to.be.true;
+    });
+    it('should not have active class if is_settings_visible is false', () => {
+        const wrapper = shallow(<ToggleSettings is_settings_visible={false} />);
+        expect(wrapper.find('.active').exists()).to.be.false;
+    });
+    it('should contain IconSettings', () => {
+        const wrapper = shallow(<ToggleSettings />);
+        expect(wrapper.contains(<IconSettings className='footer-icon' />)).to.be.true;
+    });
+    it('should have CSSTransition', () => {
+        const wrapper = shallow(<ToggleSettings is_settings_visible={true} />);
+        expect(wrapper.find(CSSTransition).exists()).to.be.true;
+    });
+    it('should have property \'in\' equal to true in CSSTransition when is_settings_visible is true', () => {
+        const wrapper = shallow(<ToggleSettings is_settings_visible={true} />);
+        expect(wrapper.find(CSSTransition).prop('in')).to.be.true;
+    });
+    it('should have property \'in\' equal to true in CSSTransition when is_settings_visible is false', () => {
+        const wrapper = shallow(<ToggleSettings is_settings_visible={false} />);
+        expect(wrapper.find(CSSTransition).prop('in')).to.be.false;
```

Comment:
I think function name should be updated to `should not have property 'in' equal to true` or `should have property 'in' equal to false`.

Reply:
But if we merge both test cases together, `property 'in' should depend on 'is_settings_visible'`.